### PR TITLE
Pass meta description placeholder to editor

### PR DESCRIPTION
**Pull request by danielFangstrom**
## Summary

This PR can be summarized in the following changelog entry:

* Adds a meta description placeholder text

## Relevant technical choices:

* Adds a descriptionEditorFieldPlaceholder prop to the snippet editor that is passed along to the draftJS editor (the name of the variable is very long and can be changed but was specified in this way in order to avoid confusion with the already existing variable `descriptionPlaceholder`)
* The color of the placeholder text is currently a lighter shade of grey compared to how it is in the old snippet editor. An [issue](https://github.com/Yoast/wordpress-seo/issues/9874) has been created for that.

## Test instructions

This PR can be tested by following these steps:

* Empty your title and meta description templates in SEO > Search Appearance > Content Types
* Go to the snippet editor, edit snippet, and empty the meta description field
* There should be a placeholder text that says "Modify your meta description by editing it right here"
* Changing focus from the meta description field should cause the text color of the placeholder to be a darker grey
* Empty the title field and save the post/page
* The text in the title field should be %%title%% - %%sitename%%
* The title field text should act like normal text, i.e. be editable
* Test these functionalities for both posts and pages

Fixes https://github.com/Yoast/wordpress-seo/issues/9830

### CODE REVIEW

File: composites/Plugin/SnippetEditor/components/SnippetEditor.js [hunk 1]
```diff
@@ -37,6 +37,8 @@ const CloseEditorButton = SnippetEditorButton.extend`
 	margin-left: 20px;
 `;
 
+const MetaDescriptionPlaceholder = "Modify your meta description by editing it right here";
```
Comment (igorschoester) [inline composites/Plugin/SnippetEditor/components/SnippetEditor.js]: Why not put this as the default in defaultProps? That way people can override it without editing this file.

File: composites/Plugin/SnippetEditor/components/SnippetEditorFields.js [hunk 2]
```diff
@@ -71,6 +71,19 @@ const InputContainer = styled.div.attrs( {
 		background-size: 25px;
 		content: "";
 	}
+	
+	&::webkit-input-placeholder {
+		color: #72777c ! important;
```
Comment (igorschoester) [inline composites/Plugin/SnippetEditor/components/SnippetEditorFields.js]: Is the ! important really needed here?

**Comment by boblinthorst** [comment]
CR: ok 👍

**Comment by boblinthorst** [comment]
No code apparently needed to be changed for the title template part of the issue. It was already fixed by another change in the code.

**Comment by boblinthorst** [comment]
Acceptance: right now, it only puts the placeholder there on save. In the previous snippet editor, a save wasn't necessary for this. This is a change compared to the previous one. This was decided by Omar.

**Pull request merged**